### PR TITLE
Add base_model_name to create/edit classifier modals with validation

### PR DESCRIPTION
**Pull request by niole**
- adds base model name to the create/edit classifier modals
- validates the model name by trying to load the spec via bumblebee
- adds dialyzer and pre-commit hooks setup to verify types
- add e2e tests for create/edit UI changes

UI Changes

Screenshot from 2024-10-19 11-53-31
Screenshot from 2024-10-19 11-53-53
Screenshot from 2024-10-19 11-53-45

### CODE REVIEW

File: lib/excision/excisions.ex [hunk 1]
```diff
@@ -460,7 +462,7 @@ defmodule Excision.Excisions do
         %{active_classifier_id: classifier.id}
       )
 
-    {:ok, _} = update_classifier(classifier, %{promoted_at: DateTime.utc_now()})
+    update_classifier(classifier, %{promoted_at: DateTime.utc_now()})
```
Comment (niole) [inline lib/excision/excisions.ex:465]: fix for type error: a consumer of promote_classifier wanted to match on :error, but promote_classifier would never return an error

File: lib/excision_web/live/classifier_live/show.ex [hunk 2]
```diff
@@ -177,7 +177,7 @@ defmodule ExcisionWeb.ClassifierLive.Show do
             "n/a"
 
           _ ->
-            Timex.Duration.from_seconds(seconds_per_iter * (max_iters - iter))
+            Timex.Duration.from_seconds(round(seconds_per_iter * (max_iters - iter)))
```
Comment (niole) [inline lib/excision_web/live/classifier_live/show.ex:180]: fix for type error: from_seconds only accepts int, but we were passing a float

File: .githooks/pre-commit [hunk 3]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+
+# run dialyzer for checking types
+mix dialyzer
```
Comment (niole) [inline .githooks/pre-commit:4]: Runs type checking when user commits code

File: test/excision_web/live/classifier_live_test.exs [hunk 4]
```diff
@@ -134,5 +172,37 @@ defmodule ExcisionWeb.ClassifierLiveTest do
       assert html =~ "Classifier updated successfully"
       assert html =~ "some updated name"
     end
+
+    test "fails to update classifier with error message if base model not findable", %{
```
Comment (feynmanliang) [inline test/excision_web/live/classifier_live_test.exs:176]: 🙏 thank you!

**Review by feynmanliang — APPROVED** [review]
Lgtm ship it

**Pull request merged**